pysb/simulator/base.py

```diff
@@ -73,9 +75,17 @@ class Simulator(object):
     """
     __metaclass__ = ABCMeta
 
+    _supports = { 'multi_initials' : False,
+                  'multi_param_values' : False }
+
     @abstractmethod
     def __init__(self, model, tspan=None, initials=None,
                  param_values=None, verbose=False, **kwargs):
+        # Get or create base PySB logger exists
+        self._logger = get_logger(self.__module__)
```

Comment:
You should use a custom `logging.LoggerAdapter` subclass here that takes `model.name` as an extra constructor arg, and implements the `process` method, returning `('[%s] %s' % (self.name, msg), kwargs)`. Then all calls to that logger will automatically prepend `[model_name]` instead of having to remember to do that explicitly at all call sites!